File: demo.in-jpaentity-med-validation/src/main/kotlin/com/example/demo/user_interface/constraints/UniqueKupeNummer.java

package com.example.demo.user_interface.constraints;

import javax.validation.Constraint;
import javax.validation.Payload;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

@Constraint(validatedBy = UniqueKupeNummerConstraintValidator.class)
@Retention(RetentionPolicy.RUNTIME)
public @interface UniqueKupeNummer {
    String message() default "Kupene må ha unike nummer i en vogn";
    Class<?>[] groups() default {};
    Class<? extends Payload>[] payload() default {};
}

File: demo.in-jpaentity-med-validation/src/main/kotlin/com/example/demo/user_interface/constraints/UniqueKupeNummer.kt
package com.example.demo.user_interface.constraints

import javax.validation.Constraint
import kotlin.reflect.KClass

@Constraint(validatedBy = [UniqueKupeNummerConstraintValidator::class])
@kotlin.annotation.Retention(AnnotationRetention.RUNTIME)
annotation class UniqueKupeNummer(
        @Suppress("unused") val message: String = "Kupene må ha unike nummer i en vogn",
        @Suppress("unused") val groups: Array<KClass<out Any>> = [],
        @Suppress("unused") val payload: Array<KClass<out Any>> = []
)
